tests/foreman/api/test_user.py

```diff
@@ -981,3 +981,100 @@ def test_expired_personal_access_token(self):
         :CaseImportance: Medium
 
         """
+
+
+@pytest.mark.no_containers
+@pytest.mark.parametrize('admin_enable', [True, False])
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+def test_positive_invalidate_users_tokens(
+    target_sat, request, admin_enable, module_org, module_location, rhel_contenthost, module_activation_key
+):
+    """Verify invalidating single and multiple users tokens.
+
+    :id: ee45cd69-d993-494c-8a14-c977096c1f52
+
+    :steps:
+        1. Create an admin user and a non-admin user with "edit_users" and "register_hosts" permission.
+        2. Generate a token with admin user and register a host with it, it should be successful.
+        3. Invalidate the token and try to use the generated token again to register the host, it should fail.
+        4. Invalidate tokens for multiple users with "invalidate-multiple" command, it should invalidate all the tokens for provided users.
+        5. Repeat Steps 2,3 and 4 with non-admin user and it should work the same way.
+
+    :expectedresults: Tokens invalidated cannot be used for registration of hosts.
+
+    :CaseImportance: Critical
+
+    :Verifies: SAT-30383
+    """
+    admin_login = gen_string('alpha')
+    non_admin_login = gen_string('alpha')
+    password = gen_string('alpha')
+
+    admin_user = target_sat.api.User(
+        login=admin_login,
+        password=password,
+        organization=[module_org],
+        location=[module_location],
+        admin=True,
+    ).create()
+    user = admin_user
+    login = admin_login
+
+    # Non-Admin user with "edit_users" permission and "Register hosts" role
+    roles = [target_sat.api.Role().create()]
+    host_register_role = target_sat.api.Role().search(query={'search': 'name="Register hosts"'})[0]
+    roles.append(host_register_role)
+    user_permissions = {
+        'User': ['edit_users'],
```

Comment:
Can we create admin_user for admin paramerter and vice versa, it seems we're creating both users twice in this case and I think just to simplify this little bit, Can we use just user variable here instead of specifying admin_user and non_admin_user explicitely,
```
password = gen_string('alpha')
if auth_type == 'admin':
    user = module_target_sat.cli_factory.user(
        {
            'login': gen_string('alpha'),
            'password': password,
            'admin': '1',
            'organization-ids': module_org.id,
        }
    )
else: 
    user = module_target_sat.cli_factory.user(
        {'login': gen_string('alpha'), 'password': password, 'organization-ids': module_org.id}
    )
    # Create role and assign required permissions here
```

Reply:
We would need both users, but as we have discussed, I have updated the code.